src/Umbraco.AuthorizedServices/Controllers/AuthorizedServiceController.cs

```diff
@@ -52,35 +52,43 @@ public class AuthorizedServiceController : BackOfficeNotificationsController
     {
         ServiceDetail serviceDetail = _serviceDetailOptions.Get(alias);
 
-        bool tokenExists = _tokenStorage.GetToken(alias) != null;
+        bool isAuthorized = CheckAuthorizationStatus(serviceDetail);
 
         string? authorizationUrl = null;
-        if (!tokenExists)
+        if (serviceDetail.AuthenticationMethod == AuthenticationMethod.OAuth2)
         {
-            AuthorizationPayload authorizationPayload = _authorizedServiceAuthorizationPayloadBuilder.BuildPayload();
+            bool tokenExists = _tokenStorage.GetToken(alias) != null;
```

Comment:
Don't think you need this call, looks like you can just re-use the `isAuthorized` value (i.e. if this is false, then the token doesn't exist).